packages/babel-plugin-transform-es2015-block-scoping/src/index.js

```diff
@@ -538,10 +538,20 @@ class BlockScoping {
     let fnPath;
     if (this.loop) {
       const ref = this.scope.generateUidIdentifier("loop");
-      const p = this.loopPath.insertBefore(
+      let transformedToBlockStatement = false;
+      if (this.loopPath.isStatementOrBlock()) {
+        transformedToBlockStatement = true;
+      }
+
+      let p = this.loopPath.insertBefore(
         t.variableDeclaration("var", [t.variableDeclarator(ref, fn)]),
       );
 
+      if (transformedToBlockStatement) {
+        p = [p[0].get("body.0")];
+        this.loopPath.requeue();
```

Comment:
Is requeue necessary?

Reply:
Yes otherwise we end up having untransformed code in the output. I would move the requeue into insertBefore/insertAfter when #5710 is merged.